ctapipe/reco/tests/test_reconstruction_methods.py

```diff
@@ -64,26 +70,53 @@ def test_reconstructors(reconstructors):
 
             try:
                 moments = hillas_parameters(geom[mask], dl1.image[mask])
-                hillas_dict[tel_id] = moments
             except HillasParameterizationError as e:
-                print(e)
+                event.dl1.tel[tel_id].parameters.hillas = HillasParametersContainer()
                 continue
 
+            # Make sure we provide only good images for the test
+            if (
+                np.isnan(moments.width.value)
+                or (moments.width.value == 0)
+            ):
+                event.dl1.tel[tel_id].parameters.hillas = HillasParametersContainer()
+            else:
+                event.dl1.tel[tel_id].parameters.hillas = moments
+
```

Comment:
here these should all be  just `dl1.parameters.hillas`, since you are already looping over `event.dl1.tel.items()`, and dl1 is the container in that dict.